packages/accounts/src/pages/login/index.page.jsx

```diff
@@ -44,30 +44,38 @@ const Page = () => {
       {({ values, errors, touched, setFieldValue, setErrors, setTouched }) => (
         <Form>
           <FormikOption fieldName={loginType.name}>
-            {({ option }) => (
-              <>
-                <FormItem name={phone.name} hidden={option !== LOGIN_TYPE.VERIFY_CODE}>
-                  <Input prefix={<PhoneInputPrefix>+86</PhoneInputPrefix>} {...phone} size="large" />
-                </FormItem>
-                <FormItem name={verifyCode.name} hidden={option !== LOGIN_TYPE.VERIFY_CODE}>
-                  <VerifyInput
-                    {...verifyCode}
-                    sendVerifyCode={() => sendVerifyCode(values[phone.name], { setErrors, setTouched })}
-                    buttonDisabled={errors[phone.name] || !values[phone.name]}
-                    size="large"
-                  />
-                </FormItem>
-                <FormItem name={identifier.name} hidden={option !== LOGIN_TYPE.PASSWORD}>
-                  <Input {...identifier} size="large" />
-                </FormItem>
-                <FormItem name={password.name} hidden={option !== LOGIN_TYPE.PASSWORD}>
-                  <Input.Password {...password} size="large" />
-                </FormItem>
-                <ActionLink onClick={() => setFieldValue(loginType.name, 1 - option)}>
-                  {LOGIN_TYPE_NAME[1 - option]}
-                </ActionLink>
-              </>
-            )}
+            {({ option }) =>
+              option === LOGIN_TYPE.VERIFY_CODE ? (
+                <>
+                  <FormItem name={phone.name}>
+                    <Input prefix={<PhoneInputPrefix>+86</PhoneInputPrefix>} {...phone} size="large" />
+                  </FormItem>
+                  <FormItem name={verifyCode.name}>
+                    <VerifyInput
+                      {...verifyCode}
+                      sendVerifyCode={() => sendVerifyCode(values[phone.name], { setErrors, setTouched })}
+                      buttonDisabled={errors[phone.name] || !values[phone.name]}
+                      size="large"
+                    />
+                  </FormItem>
+                  <ActionLink onClick={() => setFieldValue(loginType.name, 1 - option)}>
+                    {LOGIN_TYPE_NAME[1 - option]}
+                  </ActionLink>
+                </>
+              ) : (
+                <>
+                  <FormItem name={identifier.name}>
+                    <Input {...identifier} size="large" />
+                  </FormItem>
+                  <FormItem name={password.name}>
+                    <Input.Password {...password} size="large" />
+                  </FormItem>
+                  <ActionLink onClick={() => setFieldValue(loginType.name, 1 - option)}>
+                    {LOGIN_TYPE_NAME[1 - option]}
+                  </ActionLink>
```

Comment:
Alternatively, the ActionLink appears in both branch logic, so can refactor like:

```js
{option === LOGIN_TYPE.VERIFY_CODE && (
  <>
    <FormItem name={phone.name}>
      <Input prefix={<PhoneInputPrefix>+86</PhoneInputPrefix>} {...phone} size="large" />
    </FormItem>
    <FormItem name={verifyCode.name}>
       <VerifyInput
         {...verifyCode}
           sendVerifyCode={() => sendVerifyCode(values[phone.name], { setErrors, setTouched })}
           buttonDisabled={errors[phone.name] || !values[phone.name]}
           size="large"
         />
    </FormItem>
  </>
)

option !== LOGIN_TYPE.VERIFY_CODE && (
  <>
    <FormItem name={identifier.name}>
      <Input {...identifier} size="large" />
    </FormItem>
    <FormItem name={password.name}>
        <Input.Password {...password} size="large" />
    </FormItem>
  </>
)
<ActionLink onClick={() => setFieldValue(loginType.name, 1 - option)}>
  {LOGIN_TYPE_NAME[1 - option]}
</ActionLink>
```